RedHatInsights/insights-advisor-frontend
# fix(RHINENG-8460): refactor of edge&conventional devices queries and fix for filters

# Description
There was a race condition in the table caused by the fact that we were creating filter chips too early without waiting for the edge devices

Associated Jira ticket: # (issue)
https://issues.redhat.com/browse/RHINENG-8460

What I did in this PR:
I refactored the way how we request the data about edge and conventional systems. I removed try catch and now we are using the RTK query hook and passing it through the context.
Now the filter builder is requesting the data based on the edge and conventional systems requests that are resolved.
The number of requests should stay the same way as before - we request 2 endpoints for data on the edge and conventional devices, and then based on the presence/absence of these devices we make filters.
I fixed the tests for the Rules table because they were commented out.
I updated Cypress Library because the new version is faster.

How to test:
There are different scenarios:
Scenario 1: Open the table, add a filter, copy the URL, and open it in a new tab - the filter should be present, chips should be generated and filtered results should be the same
Scenario 2: Open the table from the dashboard links - the table should be opened with the filter pre-applied.
Scenario 3: Force edge devices to be false/true and repeat scenarios 1 and 2
